### DISCUSSION TITLE
Is there an easy way to import/access highlighted code in the 'Listing' Component to a plugin?

### DISCUSSION THREAD

**Discussion by bliu1013** [NationalSecurityAgency/ghidra#4506] [Q&A] [answered]
I'm new to plugin development, and I'm working on a project that needs to be able to select sections of pcode in the Listing component and bring them into a plugin component, similar to how the Decompiler can show its output for highlighted chunks of code. What would the best method to bring this information into the plugin?

**Comment by dragonmacher** [comment, marked as answer]
The short answer to this question is no.   The plugin system uses selections and highlights to communicate which addresses are active in the GUI.  However, there is no way to query the Listing for the current display elements from another plugin.

In your case, to get the selected pcode, you would essentially have to listen for selection changes and then mimic what the `PcodeFieldFactory` does to generate the visible pcode elements.    For example, for each address in the selection, you would have to to call  `instr.getPcode(true)`.     This is assuming you wanted the actual pcode objects.   If you wanted the text version of them, as displayed in the Listing, then you would have to more closely mimic the  `PcodeFieldFactory`.

**Reply by bliu1013** [reply]
That helps a lot, thanks!
